### PR TITLE
[MOD-103][Fix] Check provider permissions

### PR DESCRIPTION
**Pull request by laurenbarker**
Add additional permissions check to catch moderators trying to access a provider of which they do not have permissions.

### CODE REVIEW

File: app/routes/preprints/provider.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ export default Ember.Route.extend({
     },
 
     afterModel(model, transition) {
-        if (!model.get('reviewsWorkflow') && transition.targetName !== 'preprints.provider.setup') return this.replaceWith('preprints.provider.setup', model);
+        if (!model.get('permissions').contains('view_submissions')) {
+            this.replaceWith('forbidden');
+        } else if (!model.get('reviewsWorkflow') && transition.targetName !== 'preprints.provider.setup') {
+            return this.replaceWith('preprints.provider.setup', model);
```
Comment (aaxelb) [inline app/routes/preprints/provider.js]: `replaceWith` returns the transition object... does returning that from `afterModel` do anything?

Comment (laurenbarker) [reply app/routes/preprints/provider.js]: Yup, `afterModel` is called with a transition object. See [Transitions docs](https://guides.emberjs.com/v2.15.0/routing/preventing-and-retrying-transitions/#toc_aborting-transitions-within-code-model-code-code-beforemodel-code-code-aftermodel-code) or [Redirection docs](https://guides.emberjs.com/v2.15.0/routing/redirection/).

Comment (aaxelb) [reply app/routes/preprints/provider.js]: I'm wondering why there's a `return` here, but not two lines up. It [looks like](https://www.emberjs.com/api/ember/2.15/classes/Ember.Route/methods/afterModel?anchor=afterModel) values returned from `afterModel` are ignored unless they're promises.

Comment (laurenbarker) [reply app/routes/preprints/provider.js]: Ohhh, I get what you're saying now. I didn't add that second line but it does work. I guess it be useful for the transition to wait for it to finish.

Comment (aaxelb) [reply app/routes/preprints/provider.js]: I'm pretty sure returning does nothing, and I feel like it's misleading to include. Would you mind removing it?

**Review by aaxelb — DISMISSED** [review]

**Comment by laurenbarker** [comment]
Removed and tested 👍

**Pull request merged**